metaflow/plugins/argo/argo_workflows.py

```diff
@@ -670,24 +674,49 @@ def _compile_workflow_template(self):
                         **(
                             {
                                 # workflow status maps to Completed
-                                "notify-on-success": LifecycleHook()
+                                "slack-notify-on-success": LifecycleHook()
                                 .expression("workflow.status == 'Succeeded'")
-                                .template("notify-on-success"),
+                                .template("slack-notify-on-success"),
+                            }
+                            if self.notify_on_success and self.notify_slack_webhook_url
+                            else {}
+                        ),
+                        **(
+                            {
+                                # workflow status maps to Completed
+                                "pagerduty-notify-on-success": LifecycleHook()
+                                .expression("workflow.status == 'Succeeded'")
+                                .template("pagerduty-notify-on-success"),
                             }
                             if self.notify_on_success
+                            and self.notify_pagerduty_integration_key
                             else {}
                         ),
                         **(
                             {
                                 # workflow status maps to Failed or Error
-                                "notify-on-failure": LifecycleHook()
+                                "slack-notify-on-failure": LifecycleHook()
                                 .expression("workflow.status == 'Failed'")
-                                .template("notify-on-error"),
-                                "notify-on-error": LifecycleHook()
+                                .template("slack-notify-on-error"),
+                                "slack-notify-on-error": LifecycleHook()
                                 .expression("workflow.status == 'Error'")
-                                .template("notify-on-error"),
+                                .template("slack-notify-on-error"),
+                            }
+                            if self.notify_on_error and self.notify_slack_webhook_url
+                            else {}
+                        ),
+                        **(
+                            {
+                                # workflow status maps to Failed or Error
+                                "pagerduty-notify-on-failure": LifecycleHook()
+                                .expression("workflow.status == 'Failed'")
+                                .template("pagerduty-notify-on-error"),
+                                "pagerduty-notify-on-error": LifecycleHook()
+                                .expression("workflow.status == 'Error'")
+                                .template("pagerduty-notify-on-error"),
```

Comment:
minor nit - `notify-pagerduty-on-error` etc. to be consistent with `notify-pagerduty-integration-key`